src/python/tests/test_configuration_manager.py

```diff
@@ -0,0 +1,710 @@
+"""
+The following tests the configuration manager component For the most part,
+the code uses the boolean type for testing due to simplicity; but contains
```

Comment:
@praveen-garg - what do you mean by verification of type?

Reply:
if type is JSON, make sure non son value is not allowed; and same for all allowed types. **test coverage**

Reply:
@praveen-garg - According to @ashwinscale, the verification of the default value is done on the component, rather than when creating or updating a category. 

Example: For `_DEFAULT_PURGE_CONFIG`  it is up to the PURGE script to verify that the default/value given are a string of type int, and not something else. 
`_DEFAULT_PURGE_CONFIG = { 'iage': { 'description'='age of data', 'type'='integer', 'default'='72'}}` 


As of now, I'm not aware of process that actually check's its own configuration, in which case either everyone has to implement it, or there is (another) bug in the configuration_manager.

Reply:
I believe this must be responsibility of conf manager to make sure, data value is as per allowed types specs!
Open a TICKET please, and let's discuss there as an improvement to implement or won't fix